lib/temporal/workflow/history/event_target.rb

```diff
@@ -32,7 +33,7 @@ class UnexpectedEventType < InternalError; end
           'SIGNAL_EXTERNAL_WORKFLOW_EXECUTION'         => EXTERNAL_WORKFLOW_TYPE,
           'EXTERNAL_WORKFLOW_EXECUTION_CANCEL'         => CANCEL_EXTERNAL_WORKFLOW_REQUEST_TYPE,
           'REQUEST_CANCEL_EXTERNAL_WORKFLOW_EXECUTION' => CANCEL_EXTERNAL_WORKFLOW_REQUEST_TYPE,
-          'UPSERT_WORKFLOW_SEARCH_ATTRIBUTES'          => WORKFLOW_TYPE,
+          'UPSERT_WORKFLOW_SEARCH_ATTRIBUTES'          => UPSERT_SEARCH_ATTRIBUTES_REQUEST_TYPE,
```

Comment:
I confess, I don't really get what's going on with TARGET_TYPES or what I'm supposed to do.  @antstorm is this the code you thought might end up useful but hasn't yet?  
Overall, would love to see an explanatory comment framing this code if we want to keep it.

Reply:
The main idea here is to aggregate Temporal events around some target. This is then used to dispatch local events via the `Dispatcher` (where you need to use the target that is expected to receive those events).

I'll find some time to add comments to this class

Reply:
Yeah, it's a bit weird that sometimes the target is an object and sometimes a request.  Something feels not quite right.

Reply:
Yeah, it's either you split the events by target or make the state machine much more complicated. For example: cancellation request itself can fail, so if you were to track it on the activity/workflow target you'd need to somehow accommodate 2 parallel state machines